docs/tsp/typeServerProtocol.ts

```diff
@@ -241,36 +269,35 @@ export namespace TypeServerProtocol {
         // First request to initialize the type server.
         Initialize = 'typeServer/initialize',
         // Request from client to shut down the type server.
-        Shutdown = 'typeServer/shutdown',
+        ShutDown = 'typeServer/shutdown',
         // Request from client to get the current snapshot of the type server.
         // A snapshot is a point-in-time representation of the type server's state, including all loaded files and their types.
         // A type server should change its snapshot whenever any type it might have returned is no longer valid. Meaning types are
         // only usable for the snapshot they were returned with.
+        //
+        // Snapshots are not meant to survive any changes that would make the type server throw away its internal cache. They are merely an
+        // identifier to indicate to the client that the type server will accept requests for types from that snapshot.
         GetSnapshot = 'typeServer/getSnapshot',
```

Comment:
when the request is for a snapshot that is no longer valid, I imagine we will return undefined for every request. How will the client know to ask for a new snapshot?

Is the internal state only meant to be modified from TSP messages like `typeServer/filesChanged` (and that is when the client asks for a new snapshot)?

Pyrefly's state allows for multiple clients to connect to it with multiple "unsaved changes" states but only a single truth for state on-disk (@neil confirmation requested).

Additionally, our internal build of pyrefly can use other file watchers like watchman that might support different file systems. 

I wonder if we will need a way to push snapshot updates from the type server.

Reply:
Oh that sounds like a good idea. A notification that suggests snapshot changed. I'll add that to the notification list.